src/macros.rs

```diff
@@ -73,7 +73,7 @@ macro_rules! s {
     (it: $(#[$attr:meta])* pub struct $i:ident { $($field:tt)* }) => (
         __item! {
             #[repr(C)]
-            #[cfg_attr(feature = "extra_traits", derive(Debug, Eq, Hash, PartialEq))]
+            #[cfg_attr(feature = "extra_traits", derive(core::fmt::Debug, core::cmp::Eq, core::hash::Hash, core::cmp::PartialEq))]
```

Comment:
According to CI this doesn't work on Rust 1.13.

Reply:
Hm, that's weird. I don't know what the problem is with this syntax. I guess I'll have to install rustc 1.13 and experiment with it.

Reply:
These are the errors:

```
error: expected `,`, found `::`
  --> src/macros.rs:76:61
   |
76 |             #[cfg_attr(feature = "extra_traits", derive(core::fmt::Debug, core::cmp::Eq, core::hash::Hash, core::cmp::PartialEq))]
   |                                                             ^^

error: expected `,`, found `::`
  --> src/macros.rs:76:61
   |
76 |             #[cfg_attr(feature = "extra_traits", derive(core::fmt::Debug, core::cmp::Eq, core::hash::Hash, core::cmp::PartialEq))]
   |                                                             ^^

error: expected `]`, found `::`
  --> src/macros.rs:76:61
   |
76 |             #[cfg_attr(feature = "extra_traits", derive(core::fmt::Debug, core::cmp::Eq, core::hash::Hash, core::cmp::PartialEq))]
   |                                                             ^^

error: Could not compile `libc`.
```

I have no idea what the problem is and how to work around this. Anyone have any ideas?

Reply:
I guess qualified paths were not allowed in `derive` arguments in 1.13.

When building libc for rustc we know we're using a more recent version, so ideally I wouldn't have to deal with this.

Reply:
We can use `cfg(feature = "rustc-dep-of-std")` for it.

Reply:
I don't understand how that would work. The `core::fmt::...` syntax will still be there and rustc 1.13 will try to parse it, no?

Reply:
That syntax won't work on 1.13: https://rust.godbolt.org/z/f9nf9rTTG

So, we want to make it work under `rustc-dep-of-std`, right? Then we can hide them behind `rustc-dep-of-std`, it's fine because we suppose it's used on rust-lang/rust.